synapse/types.py

```diff
@@ -51,7 +51,15 @@ class Collection(Iterable[T_co], Container[T_co], Sized):  # type: ignore
 
 class Requester(
```

Comment:
tempted to say that we should take the opportunity to convert this to an `@attr.s`, but I guess we shouldn't scope-creep this more than it has already.

Reply:
I think that would be a good change, I think we should do that with a bunch of the namedtuples we have throughout the code-base though. Seems like a good refactoring to start after the async/await work.